psi4/src/psi4/optking/molecule_read_coords.cc

```diff
@@ -567,12 +567,24 @@ bool stoi(string s, int *a) {
 
 // convert string to float
 bool stof(string s, double *val) {
-  double i = atof(s.c_str());
-  if (i!=0) {
-    *val = i;
+   double d;
+   std::istringstream iss(s);
+   iss >> d;
+   if (iss.bad() || iss.fail())
```

Comment:
Just out of curiosity: wouldn't std::stod() be more convenient/compact than the istringstream solution?